lib/mayaUsd/commands/layerEditorCommand.cpp

```diff
@@ -875,6 +882,110 @@ class LockLayer : public BaseCmd
     Ufe::Selection _savedSn;
 };
 
+class RefreshSystemLockLayer : public BaseCmd
+{
+public:
+    RefreshSystemLockLayer()
+        : BaseCmd(CmdId::kRefreshSystemLock)
+    {
+    }
+
+    bool doIt(SdfLayerHandle layer) override
+    {
+        auto stage = getStage();
+        if (!stage)
+            return false;
+
+        if (_refreshSubLayers) {
+            // If refreshSubLayers is True, we attempt to refresh the system lock status of all
+            // layers under the given layer. This is specially useful when reloading a stage.
+            bool includeTopLayer = true;
+            auto allLayers = MayaUsd::getAllSublayerRefs(layer, includeTopLayer);
+            for (auto layerIt : allLayers) {
+                _refreshLayerSystemLock(layerIt);
+            }
+        } else {
+            // Only check and refresh the system lock status of the current layer.
+            _refreshLayerSystemLock(layer);
+        }
+
+        return true;
+    }
+
+    // The command itself doesn't retain its state. However, the underlying logic contains commands
+    // that are undoable.
+    bool undoIt(SdfLayerHandle layer) override { return true; }
+
+    std::string _proxyShapePath;
+    bool        _refreshSubLayers = false;
+
+private:
+    std::string _quote(const std::string& string)
+    {
+        return std::string(" \"") + string + std::string("\"");
```

Comment:
std:: in C++has the [std::quoted function](https://en.cppreference.com/w/cpp/io/manip/quoted) which deals with nested quotes. No need to update just for this.

Reply:
Good advice! std::quoted looks a lot cleaner. I'll be using that in the upcoming PR.